src/services/node-checker-wrapper.ts

```diff
@@ -0,0 +1,272 @@
+import { Redis } from 'ioredis'
+import { Pocket, Node, PocketAAT, Configuration, Session } from '@pokt-network/pocket-js'
+import { getNodeNetworkData, removeNodeFromSession } from '../utils/cache'
+import { MAX_RELAYS_ERROR } from '../utils/constants'
+import { hashBlockchainNodes } from '../utils/helpers'
+import { MetricsRecorder } from './metrics-recorder'
+import { ArchivalCheck, ChainCheck, Check, NodeChecker, NodeCheckResponse, SyncCheck } from './node-checker'
+
+const logger = require('../services/logger')
+
+export class NodeCheckerWrapper {
+  pocket: Pocket
+  redis: Redis
+  metricsRecorder: MetricsRecorder
+  origin: string
+
+  constructor(pocket: Pocket, redis: Redis, metricsRecorder: MetricsRecorder, origin: string) {
+    this.pocket = pocket
+    this.redis = redis
+    this.metricsRecorder = metricsRecorder
+    this.origin = origin
+  }
+
+  /**
+   * Helper method to check for cached checks, is a check is already cached or in progress,
+   * returns valid nodes from the cached result or session. Otherwise set a cache lock.
+   * @param {Node[]} nodes session nodes.
+   * @param cacheKey key to get/set results.
+   * @returns nodes cached or provided on case of cache/lock.
+   */
+  protected async cacheNodes(nodes: Node[], cacheKey: string): Promise<Node[]> {
+    const checkedNodes: Node[] = []
+    let checkedNodesList: string[] = []
+
+    const checkedNodesCached = await this.redis.get(cacheKey)
+
+    checkedNodesList = JSON.parse(checkedNodesCached)
+
+    if (checkedNodesList && checkedNodesList.length > 0) {
+      for (const node of nodes) {
+        if (checkedNodesList.includes(node.publicKey)) {
+          checkedNodes.push(node)
+        }
+      }
+      return checkedNodes
+    }
+
+    // Cache is stale, start a new cache fill
+    // First check cache lock key; if lock key exists, return full node set
+    const chainLock = await this.redis.get('lock-' + cacheKey)
+
+    if (chainLock) {
+      return nodes
+    } else {
+      // Set lock as this thread checks the Chain with 60 second ttl.
+      // If any major errors happen below, it will retry the check every 60 seconds.
+      await this.redis.set('lock-' + cacheKey, 'true', 'EX', 60)
+    }
+
+    return checkedNodes
+  }
+
+  /**
+   * Logs responses from the node checks and filters failing nodes.
+   * @param checkType type of check made.
+   * @param nodes nodes to be filtered.
+   * @param nodesPromise results of a check on node's.
+   * @param blockchainID blockchain used for the checks.
+   * @param requestID request id.
+   * @param relayStart time when the checks started.
+   * @param applicationID application database's ID.
+   * @param applicationPublicKey application's public key.
+   * @returns nodes having successful and valid check results.
+   */
+  protected async filterNodes<T>(
+    checkType: Check,
+    nodes: Node[],
+    nodesPromise: PromiseSettledResult<NodeCheckResponse<T>>[],
```

Comment:
This variable name could be a little bit more descriptive—the "type" of it doesn't need to be in the name, as it can be inferred from its typescript type.

Reply:
On that case it could be just `filter` then as is what it does besides logging

Reply:
> On that case it could be just `filter` then as is what it does besides logging

I think he meant the `nodesPromise` variable.